skbio/diversity/_driver.py

```diff
@@ -180,9 +182,114 @@ def alpha_diversity(metric, counts, ids=None, validate=True, **kwargs):
     return pd.Series(results, index=ids)
 
 
+@deprecated(as_of='0.4.2-dev', until='0.5.0',
+            reason=('The return type is unstable. Developer caution is '
+                    'advised. The resulting DistanceMatrix object may include '
+                    'zeros when distance has not been calculated, and '
+                    'therefore can be misleading.'))
+def partial_beta_diversity(metric, counts, ids=None, validate=None,
+                           id_pairs=None, **kwargs):
+    """Compute distances only between specified ID pairs
+
+    Parameters
+    ----------
+    metric : callable
+        The pairwise distance function to apply. ``metric`` must be resolvable
+        by scikit-bio (e.g., UniFrac methods), or must be callable.
+    counts : 2D array_like of ints or floats
+        Matrix containing count/abundance data where each row contains counts
+        of OTUs in a given sample.
+    ids : iterable of strs
+        Identifiers for each sample in ``counts``.
+    id_pairs : iterable of tuple, optional
+        An iterable of tuples of IDs to compare (e.g., ``[('a', 'b'), ('a',
+        'c'), ...])``. If specified, the set of IDs described must be a subset
+        of ``ids``.
+    validate : ignored
+        This kwarg is preserved to maintain consistency with the beta_diversity
+        signature.
+    kwargs : kwargs, optional
+        Metric-specific parameters.
+
+    Returns
+    -------
+    skbio.DistanceMatrix
+        Distances between all pairs of samples (i.e., rows). The number of
```

Comment:
change this section to:

*Distances between pairs of samples indicated by ``id_pairs``. Pairwise distances not defined by `id_pairs` will be 0.0. Use this resulting ``DistanceMatrix`` with caution as 0.0 is a valid distance.*